### DISCUSSION TITLE
custom configuration-time properties for serilog?

### DISCUSSION THREAD

**Discussion by Balfa** [datalust/seq-tickets#2312] [Help] [answered]
Splunk's serilog sink allows setting a "[source](https://github.com/serilog-contrib/serilog-sinks-splunk/blob/30ab378bbc72748fe60ac0226066dc23aa865809/src/Serilog.Sinks.Splunk/SplunkLoggingConfigurationExtensions.cs#L180C20-L180C26)", which we set to "WeatherApi", "BlogPostApi", etc. so we can store logs from all our services in one splunk index, but still easily focus on just one at a time when we're tracking down a problem.

Am I missing something or is there no easy way to do the equivalent in the seq sink? We could even use something like the [CustomFields property](https://github.com/serilog-contrib/serilog-sinks-splunk/blob/30ab378bbc72748fe60ac0226066dc23aa865809/src/Serilog.Sinks.Splunk/SplunkLoggingConfigurationExtensions.cs#L178) and set it at configuration time.

The only other solution I'm thinking of is just pushing some unique hardcoded log context at app startup, but that doesn't seem as nice, nor does it fit in well with our existing model.

**Comment by nblumhardt** [comment, marked as answer]
Hi Mo,

The best solution for this with Seq is to give each source a unique API key. Then, in the API key configuration on the Seq side, apply a property `Source = ...` to identify the events that are ingested using that API key. You can attach multiple properties to each key, so this is also where `Environment` etc. would normally be applied, if you need them.

This has a few advantages -

 * The Data > Ingestion screen will then show you the log volume coming in from each source,
 * You can suppress or filter ingestion from a single source, e.g. in the case of runaway logging, and
 * Payloads are smaller (you don't need to redundantly send the same "source" properties with every event.

Docs on API keys are in https://docs.datalust.co/docs/api-keys.

If you decide to go the other way and configure it client-side, the `Enrich.WithProperty("Source", value)` option on Serilog's `LoggerConfiguration` does what you're looking for.

Hope this helps,
Nick

**Reply by Balfa** [reply]
Thanks @nblumhardt ! This looks like just what I need. And thanks for all your FOSS contributions 🤘

**Reply by nblumhardt** [reply]
Thanks @Balfa :-) 👍
